docs/c_api.md

```diff
@@ -1,16 +1,21 @@
+The C interface is not new, but previously it was part of the `c2pa-c` repo. The C interface is useful to other bindings, so it was exported as a Rust JSON API. But that led to things like `c2pa-python` importing from `c2pa-c` and then re-exporting a Python API via [UniFFI](https://mozilla.github.io/uniffi-rs/latest/). The UniFFI tools have severe limitations in what they represent, such as the inability to have mutable parameters, and there are other limitations with the tools for binding Rust to C++, Swift, and other languages. 
 
-We still need to write bindings for each language, but since there are so many examples of this, AI engines are very good at writing the code. The result is well formed, well documented bindings. I've found it takes some manual effort to fix some things. Instead of unreadable, incomprehensible auto generated binding glue, we end up with well structured code bindings that can be customized for our needs.
+## New approach
 
-#### Tradeoffs
-The C language is not object oriented and does not natively support things like exception handling. There is no garbage collection. APIs use unsafe pointer references. We must be very careful about pointers an memory management. So the C API is not something we want developers to use directly. But it makes a very solid common way to bridge between Rust and other languages when used correctly. Higher level structures in the other languages can ensure that references to Rust structures are correctly managed and freed. 
+Binding to C APIs is a well-established and mature practice. Every language has well-documented methods for binding to C.  Rust has built in support for writing C bindings. If we provide a solid C interface, we can simply bind other languages to C and leverage all the work that has gone into those bindings.  
+
+Bindings must still be written for each language, but since there are so many examples of this, AI engines are very good at writing the code, resulting in well-formed, well-documented bindings, though some manual effort is required to fix some things. Instead of unreadable, incomprehensible auto-generated binding glue, the result is well-structured code bindings that can be customized as needed.
+
+## Tradeoffs
+
+The C language is not object-oriented, does not perform garbage collection, and does not natively support things like exception handling.  API may use unsafe pointer references, so care must be taken with pointers and memory management. 
+
+For these reasons, you shouldn't use the C API directly. But it makes a very solid common way to bridge between Rust and other languages when used correctly. Use higher-level structures in the other languages to ensure that references to Rust structures are correctly managed and freed. 
```

Comment:
```suggestion
For these reasons, you shouldn't use the C API directly. But it makes a very solid common way to bridge between Rust and other languages when used correctly. Use higher-level structures in the other languages to ensure that references to native structures are correctly managed and freed. 
```
We need to have a concept that Rust is different than C, and here we deal with C-level APIs.